src/database.ts

```diff
@@ -3222,8 +3257,7 @@ class Database extends common.GrpcServiceObject {
           span.addEvent('No session available', {
             'session.id': session?.id,
           });
-          this.runTransaction(options, runFn!);
-          span.end();
+          retry(span);
```

Comment:
Same here.

Reply:
Hmmm but the problem then is that we shall be ending the span prematurely before the code inside of database.runTransaction has exited hence why we add event listeners on the transaction.

Reply:
@surbhigarg92 where you saying that we should end the span before invoking the retry? I lost context of this comment.